src/views/ExportHistoryDropdown.tsx

```diff
@@ -0,0 +1,264 @@
+  const allFills = useSelector(getSubaccountFills, shallowEqual) ?? [];
+  const { decimal: localeDecimalSeparator, group: localeGroupSeparator } = useLocaleSeparators();
+  const [checkedTrades, setCheckedTrades] = useState(true);
+  const [checkedTransfers, setCheckedTransfers] = useState(true);
+
+  const trades = useMemo(
+    () =>
+      allFills.map((fill) => {
+        const { sign: feeSign, formattedString: feeString } = formatNumber({
+          type: OutputType.Fiat,
+          value: fill.fee,
+          localeDecimalSeparator,
+          localeGroupSeparator,
+        });
+
+        const { sign: totalSign, formattedString: totalString } = formatNumber({
+          type: OutputType.Fiat,
+          value: MustBigNumber(fill.price).times(fill.size),
+          localeDecimalSeparator,
+          localeGroupSeparator,
+        });
+
+        const { displayString } = formatTimestamp({
+          type: OutputType.DateTime,
+          value: fill.createdAtMilliseconds,
+          locale: selectedLocale,
+        });
+
+        const sideKey = {
+          [OrderSide.BUY]: STRING_KEYS.BUY,
+          [OrderSide.SELL]: STRING_KEYS.SELL,
+        }[fill.side.rawValue];
+
+        return {
+          type: fill.resources.typeStringKey && stringGetter({ key: fill.resources.typeStringKey }),
+          liquidity:
+            fill.resources.liquidityStringKey &&
+            stringGetter({ key: fill.resources.liquidityStringKey }),
+          time: displayString,
+          amount: fill.size,
+          fee: feeSign ? `${feeSign}${feeString}` : feeString,
+          total: totalSign ? `${totalSign}${totalString}` : totalString,
+          market: fill.marketId,
+          side: sideKey
+            ? stringGetter({
+                key: sideKey,
+              })
+            : '',
+        };
+      }),
+    [allFills, stringGetter, localeDecimalSeparator, localeGroupSeparator]
+  );
+
+  const transfers = useMemo(
+    () =>
+      allTransfers.map((transfer) => {
+        const { sign, formattedString } = formatNumber({
+          type: OutputType.Fiat,
+          value: transfer.amount,
+          localeDecimalSeparator,
+          localeGroupSeparator,
+        });
+
+        const { displayString } = formatTimestamp({
+          type: OutputType.DateTime,
+          value: transfer.updatedAtMilliseconds,
+          locale: selectedLocale,
+        });
+
+        return {
+          time: displayString,
+          action:
+            transfer.resources.typeStringKey &&
+            stringGetter({ key: transfer.resources.typeStringKey }),
+          sender: transfer.fromAddress,
+          recipient: transfer.toAddress,
+          amount: sign ? `${sign}${formattedString}` : formattedString,
+          transaction: transfer.transactionHash,
+        };
+      }),
+    [allTransfers, stringGetter, localeDecimalSeparator, localeGroupSeparator]
+  );
+
+  const exportTrades = useCallback(() => {
+    exportCSV(trades, {
```

Comment:
`trades` and `transfers` that you are exporting from the redux store is only a partial record. Would it be possible to loop and fetch their historical data? Maybe allow the user to specify a date range as well.

Reply:
Ye makes sense, I didn't knew that was a partial record, I'm trying to use the following services from dydx js client:

- [getSubaccountsFills](https://github.com/dydxprotocol/v4-clients/blob/37f6ed199c7f6f586632f28bfe78b25422d0be0f/v4-client-js/src/clients/modules/account.ts#L116)
- [getSubaccountTransfers](https://github.com/dydxprotocol/v4-clients/blob/37f6ed199c7f6f586632f28bfe78b25422d0be0f/v4-client-js/src/clients/modules/account.ts#L65)

But there seems to be a limit of a maximum of 100 entries returned from service.

It would be useful to have an endpoint that is able to return all data given a date range, although this for some users might result in a very large file or payload.

If I were to try to reconstruct the data by following a loop, I can request 100 elements and each time take the last element from which I then get the new height to make the query, thus going backwards until I receive an array with less than 100 elements.
The only problem with this technique is that if more than 100 operations were performed in a block, I would lose them this way.

So imho it's not the best solution, I suggest to create a single endpoint that can allow to retrieve all the data using a data range, even if as I said, in some cases, the payload can be very large